lib/bridge/QuitDebouncer.js

```diff
@@ -0,0 +1,106 @@
+/*eslint no-invalid-this: 0*/
+"use strict";
+var Promise = require("bluebird");
+
+function QuitDebouncer(ircBridge) {
+    this.ircBridge = ircBridge;
+
+    this._rejoinPromises = {
+        // '$nick $server.domain': Promise that resolves if the user joins a channel having quit
+        //       previously
+    };
+    this._rejoinResolvers = {
+        // '$nick $server.domain': function to call when a user joins
+    };
+
+    // Heuristic to measure the probability of a net-split having just happened
+    // This is to smooth incoming PART spam from IRC clients that suffer from a
+    // net-split (or other issues that lead to mass PART-ings)
+    this._recentQuits = {
+        // $server => {integer} : the number of recent parts with kind = quit
+    };
+    this._quitsPerSecond = {
+        // $server => {integer} : the number of quits in the last interval 1s
+    };
+
+    this._is_split_occuring = {
+        // $server => {bool}
+    };
+
+    setInterval(() => {
+        Object.keys(this._recentQuits).forEach(
+            (server) => {
+                this._quitsPerSecond[server] = this._recentQuits[server];
+                this._recentQuits[server] = 0;
+
+                let threshold = 5;
+
+                // Possible net-split
+                if (this._quitsPerSecond[server] > threshold) {
+                    this._is_split_occuring = true;
```

Comment:
This fails in multi-server scenarios (where an IRC bridge is bridging to Freenode and Moznet for example). If Freenode netsplits, we will incorrectly execute netsplitting logic for **all servers**, in this case including Moznet.